### PR TITLE
Begleysm's changes to make BasicMvvmSample more new-user friendly

### PR DESCRIPTION
**Pull request by begleysm**
These changes are the result of my readthrough of the BasicMvvmSample as a new user.  I added more detail & explaination where I thought it would help.  I also added text & a screenshot of the Rider New Solution screen alongside the existing Visual Studio screenshot.

### CODE REVIEW

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md [hunk 1]
```diff
@@ -32,35 +32,52 @@ The `MVVM` pattern helps us to focus on either of the three parts. If we get new
 
 ### INotifyPropertyChanged
 
-Our `View` needs any way to know when a property has changed and the `View` should update itself. To achieve this there is an interface called `INotifyPropertyChanged` which implements the event `PropertyChanged`. If our `ViewModel` implements this interface, we can send update notifications to our `View` or any other class listening to that event. 
+C# uses `Properties` to read, write, or compute the value of a private `Fields`.  C# uses `Interfaces` to define a contract that `Classes` can satisfy by providing an implementation for the members defined in the `Interface`.
```
Comment (timunie) [inline src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:35]: > ... of a private `Fields` ...

There is something wrong in that sentence.

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md [hunk 2]
```diff
@@ -32,35 +32,52 @@ The `MVVM` pattern helps us to focus on either of the three parts. If we get new
...
+- More on Properties: <a href="https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties" target="_blank">[Microsoft Docs]</a>
+- More on Fields: <a href="https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/fields" target="_blank">[Microsoft Docs]</a>
+- More on Interfaces: <a href="https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/interface" target="_blank">[Microsoft Docs]</a>
+- More on Classes: <a href="https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/class" target="_blank">[Microsoft Docs]</a>
```
Comment (timunie) [inline src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:39]: I somehow like the idea if explaining these fundamentales. But maybe we should do it one chapter before, like

```md
### Properties, Fields, Classes and Interfaces
- An `Interface` is ... [more](...)
- ...
```

what do you think?

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md [hunk 3]
```diff
@@ -32,35 +32,52 @@ The `MVVM` pattern helps us to focus on either of the three parts. If we get new
...
+
+Our `View` needs a way to know when a `Property` and that the `View` should update itself. To achieve this there is an `Interface` called `INotifyPropertyChanged` which implements the event `PropertyChanged`. If our `ViewModel` implements this interface, we can send update notifications to our `View` or any other class listening to that event.
```
Comment (timunie) [inline src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:41]: > ... when a `Property` and that the ...

 missing word?

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md [hunk 4]
```diff
@@ -32,35 +32,52 @@ The `MVVM` pattern helps us to focus on either of the three parts. If we get new
...
 
-## Solution 1 : Implement INotifyPropertyChanged on our own
+## Approach 1 : Implement INotifyPropertyChanged on our own
 
 In this sample we will show you how you can implement and use the interface `INotifyPropertyChanged` on your own. 
 
 ### Step 1: Create SimpleViewModel
 
-In our project structure we have a folder called `ViewModels`. We will add a new class called "SimpleViewModel" there. This class should implement `INotifyPropertyChanged` like shown below: 
+In our project structure we have a folder called `ViewModels`. The `ViewModels` act as an intermediary between the `View` and the `Model`. Add a new class called "SimpleViewModel" there. This class should implement `INotifyPropertyChanged` as shown below:
```
Comment (timunie) [inline src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:80]: > The `ViewModels` act as an intermediary between the `View` and the `Model`

This was explained before. Should be repeat it here?

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md [hunk 5]
```diff
@@ -178,11 +193,13 @@ We enable [[`CompiledBindings`](https://docs.avaloniaui.net/docs/data-binding/co
 </Window>
 ```
 
+Note that the lines `x:CompileBindings="True"` and `x:DataType="vm:MainWindowViewModel"` enables [[`CompiledBindings`](https://docs.avaloniaui.net/docs/data-binding/compiledbindings)] in our `Window`. `CompiledBindings` has many benefits such as better performance and a better debugging experience.
```
Comment (timunie) [inline src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:196]: I want to tell the user that this step is optionally, but highly recommended. Can you add that somehow back?

**Review by timunie — COMMENTED** [review]
Overall, I really love the ideas and changes you made. I have some (minor) things, where I think you had a typo and some things I am not sure about. Please let me know your thoughts.

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md (hunk 3 above)
Comment (begleysm) [reply src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:41]: yep, should say "when a Property has changed"

File: src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md (hunk 4 above)
Comment (begleysm) [reply src/Avalonia.Samples/MVVM/BasicMvvmSample/README.md:80]: You're right.  That could go away.

**Comment by begleysm** [comment]
I'll make some changes and submit an updated PR.

**Comment by begleysm** [comment]
I've implemented the suggested changed which should now be part of the PR.

**Comment by timunie** [comment]
LGTM 👌 

Thank you for your contribution 🌹

**Pull request merged**